src/Contracts/Http.php

```diff
@@ -0,0 +1,16 @@
+<?php
+
+namespace MeiliSearch\Contracts;
+
+interface Http
+{
+    public function get($path, array $query = []);
+
+    public function post(string $path, $body = null, array $query = []);
+
+    public function put(string $path, $body = null, array $query = []);
+
+    public function patch(string $path, $body = null, array $query = []);
+
+    public function delete($path, array $query = []);
```

Comment:
Return typehint for each method?

Reply:
The issue is that the responses can be `null`, `array`, or `string`. So the return type has to be `mixed` which I believe is same as not having a return type. Otherwise we will have to alter the response type be a DTO or something like your last comment. Which will break a lot of things in the public api. So I think the return types for these methods should be in a separate PR. with appropriate BC break labels or something, what do you think?

Reply:
I agree, maybe it could be a good idea to submit a new PR that force more strict "logic" and types, but for now, that's okay for me 🙂

Reply:
I agree, this kind of big change would be appreciated, but in another PR 🙂